src/v2/guide/composition.md

```diff
@@ -0,0 +1,546 @@
+
+This is useful when we'd really like to _remove_ options from a base interface. For example, in the case of `ErrorNote`, it'd be very strange for it to be used like this:
+
+``` html
+<error-note type="success">
+  <span slot="bold">Welcome!</span>
+  Would you like to take the tour?
+</error-note>
+```
+
+Above, a developer has hijacked the component for a new kind of note that is not actually an error. The wrapper version of this component more strictly controls the API, so does not allow this kind of abuse.
+
+<p class="tip">In the example above, <code>type="success"</code> <em>would</em> actually affect the wrapped <code>AppNote</code> component, but as an attribute rather than a prop. To prevent attributes from automatically being added to the root element of a component, use the <a href="../api/#inheritAttrs"><code>inheritAttrs: false</code></a> option.</p>
+
+
+
+## Factory Functions
+
+**Factory functions** are functions that return component options, like this:
+
+``` js
+function createComponent (options) {
+  return {
+    // ... dynamically generated component config ...
+  }
+}
+```
+
+Factory functions allow you to better control and simplify how component variations are created. For example, it could be argued that the `ErrorNote` we [created earlier with extension](#An-Example-Use-Case) would be conceptually simpler as a factory function:
+
+``` js
+function createNote (options) {
+  options = options || {}
+  options.props = options.props || {}
+  options.slots = options.slots || {}
+
+  return {
+    props: {
+      type: {
+        type: String,
+        default: options.props.type || 'info'
+      }
+    },
+    template: `
+      <p :class="['note', type]">
+        <strong>
+          <slot name="bold">${
+            options.slots.bold || ''
+          }</slot>
+        </strong>
+        <slot>${
+          options.slots.default || ''
+        }</slot>
+      </p>
+    `
+  }
+}
+```
+
+Then we could create our `AppNote` and `ErrorNote` components with:
+
+``` js
+var AppNote = createNote()
+
+var ErrorNote = createNote({
+  props: {
+    type: 'error'
+  },
+  slots: {
+    bold: 'Error!'
+  }
+})
+```
+
+Using this strategy, all notes in the app maintain unified behavior, but _you_ control the API for creating them. This can be especially useful when you want to:
+
+- Create many variations of a component, abstracting away the specific implementation details.
+- Give designers, less experienced developers, or members of a very large team the ability to create new component variations, but strictly control what they're allowed to change.
+- Dynamically build template strings at runtime, as a more familiar alternative to render functions.
+
+We'll demonstrate the last advantage in the next section.
+
+### Computed Components
+
+Imagine you want a component that generates dynamic headings, with this kind of interface:
+
+``` html
+<dynamic-heading :level="headingLevel">
+  {{ headingContent }}
+</dynamic-heading>
+```
+
```

Comment:
Yeah, I was previously including more render function code here and simplified the examples so as not to dive too deeply into render functions, but I think it should be possible now.

Reply:
I have an example with SVG avatars that I was working on, but I'm not sure if that would leave out people who don't know SVG

Reply:
@sdras That'd be something I'd worry about too, but I'd still love to see it as an example to learn from. It could possibly even be adapted into something simpler.

Reply:
Update on this, @codebryo might be up for writing a cookbook entry to do a bit more of a deep dive into this, so if we get that, we can keep this example simple and link his up for further clarification :)

Reply:
I will actually write that entry :)